src/browser/modules/Sidebar/ProjectsFilesScripts.tsx

```diff
@@ -0,0 +1,239 @@
+import { Bus } from 'suber'
+import { isWindows } from 'browser/modules/App/keyboardShortcuts'
+
+interface ProjectFilesError {
+  apolloErrors: (ApolloError | undefined)[]
+  errors?: string[]
+}
+
+export const ProjectFilesError = ({
+  apolloErrors,
+  errors
+}: ProjectFilesError): JSX.Element => {
+  const definedApolloErrors = filter(
+    apolloErrors,
+    apolloError => apolloError !== undefined
+  )
+  const definedErrors = filter(errors, error => error !== '')
+  const ErrorsList = () => {
+    let errorsList: JSX.Element[] = []
+    if (size(definedErrors)) {
+      errorsList = [
+        ...errorsList,
+        ...definedErrors.map((definedError, i) => (
+          <span key={`${definedError}-${i}`}>{definedError}</span>
+        ))
+      ]
+    }
+    if (size(definedApolloErrors)) {
+      definedApolloErrors.map((definedApolloError, j) => {
+        if (definedApolloError?.graphQLErrors.length) {
+          errorsList = [
+            ...errorsList,
+            ...definedApolloError.graphQLErrors.map(({ message }, k) => (
+              <span key={`${message}-${j}-${k}`}>{message}</span>
+            ))
+          ]
+        }
+        if (definedApolloError?.networkError) {
+          errorsList = [
+            ...errorsList,
+            <span key={j}>A network error has occurred.</span>
+          ]
+        }
+      })
+    }
+    return errorsList
+  }
+
+  return (
+    <Render if={size(definedApolloErrors) || size(definedErrors)}>
+      {ErrorsList()}
+    </Render>
+  )
+}
+
+interface ProjectFilesScripts {
+  bus: Bus
+  onSelectScript: (favorite: Favorite) => void
+  onExecScript: (favorite: Favorite) => void
+  onExportScripts: () => void
+  onUpdateFolder: () => void
+  onRemoveFolder: () => void
+  scriptsNamespace: string
+  title: string
+  projectId: string
+  relateApiToken: string
+  neo4jDesktopGraphAppId: string
+  relateUrl: string
+}
+
+function ProjectFilesScripts(props: ProjectFilesScripts): JSX.Element {
+  const { data, error: getProjectFilesError, refetch } = useQuery<
+    ProjectFilesResult,
+    ProjectFilesVariables
+  >(GET_PROJECT_FILES, {
+    variables: ProjectFilesQueryVars(props.projectId)
+  })
+  const [removeFile, { error: removeProjectFileError }] = useMutation(
+    DELETE_PROJECT_FILE
+  )
+  const [projectFiles, setProjectFiles] = useState<Favorite[]>([])
+
+  useEffect(() => {
+    let isStillMounted = true
+    if (data) {
+      const getProjectFiles = async () => {
+        const getProjectFilePromises = data.getProject.files.map(
+          ({ downloadToken, name, directory }) =>
+            mapProjectFileToFavorites({
+              downloadToken,
+              name,
+              directory,
+              apiToken: props.relateApiToken,
+              clientId: props.neo4jDesktopGraphAppId,
+              relateUrl: props.relateUrl
+            })
+        )
+        const projectFiles = await Promise.all(getProjectFilePromises)
```

Comment:
technically speaking shadowing, but wont affect logic at this point

Reply:
yeah, should name it something else ideally but Ill leave it for now